lib/hexdocs/queue.ex

```diff
@@ -250,4 +266,96 @@ defmodule Hexdocs.Queue do
       )
     end)
   end
+
+  defp find_search_items(package, version, files) do
+    search_data =
+      Enum.find_value(files, fn {path, content} ->
+        case Path.basename(path) do
+          "search_data-" <> _digest -> {:elixir, content}
+          "search-data.js" -> {:gleam, content}
+          _other -> nil
+        end
+      end)
+
+    case search_data do
+      {:elixir, content} ->
+        extract_elixir_search_items(package, version, content)
+
+      {:gleam, content} ->
+        extract_gleam_search_items(package, version, content)
+
+      nil ->
+        Logger.info("Failed to find search data for #{package} #{version}")
+        nil
+    end
+  end
+
+  defp extract_gleam_search_items(package, version, search_data) do
+    "window.Gleam.initSearch(" <> search_items_js = search_data
+
+    search_items =
+      case Jason.decode(search_items_js) do
+        {:ok, search_items} ->
+          search_items
+
+        {:error, %Jason.DecodeError{position: position}} ->
+          case Jason.decode(:binary.part(search_items_js, 0, position)) do
+            {:ok, search_items} ->
+              search_items
+
+            {:error, error} ->
+              Logger.error(
+                "Failed to decode search data json for #{package} #{version}: " <>
+                  Exception.message(error)
+              )
+
+              nil
+          end
+      end
+
+    case search_items do
+      items when is_list(items) ->
+        Enum.map(items, fn item ->
+          %{"content" => doc, "doc" => _, "title" => title, "url" => ref} = item
+          %{"doc" => doc, "title" => title, "ref" => ref}
+        end)
+
+      nil ->
+        nil
+
+      _ ->
+        Logger.error("Failed to extract search items from search data for #{package} #{version}")
+        nil
+    end
+  end
+
+  defp extract_elixir_search_items(package, version, search_data) do
+    "searchData=" <> search_data_json = search_data
+
+    search_data =
+      case Jason.decode(search_data_json) do
+        {:ok, search_data} ->
+          search_data
+
+        {:error, error} ->
+          Logger.error(
+            "Failed to decode search data json for #{package} #{version}: " <>
+              Exception.message(error)
+          )
+
+          nil
+      end
+
+    case search_data do
+      %{"items" => items} ->
+        items
```

Comment:
I have added the proglang to ExDoc just now. It means it won't be available for old ones, but we can get it like this:

```elixir
language =
  case search_data do
    %{"proglang" => proglang} =>
      proglang

    %{"items" => items} =>
      # For older versions, we infer the module
      if Enum.any?(items, &elixir_module?/1), do: "elixir", else: "erlang"
  end
```

where

```elixir
defp elixir_module?(%{type: "module", title: <<first_letter, _::binary>>}) when first_letter in ?A..?Z, do: true
defp elixir_module?(_), do: false
```